common/djangoapps/util/enterprise_helpers.py

```diff
@@ -442,3 +446,41 @@ def get_dashboard_consent_notification(request, user, course_enrollments):
             }
         )
     return ''
+
+
+def is_course_in_enterprise_catalog(site, course_id, user, enterprise_catalog_id):
+    """
+    Verify that the provided course id exists in the site base list of course
+    run keys from the provided enterprise course catalog.
+
+    Arguments:
+        course_id (str): The course ID.
+        site: (django.contrib.sites.Site) site instance
+        enterprise_catalog_id (Int): Course catalog id of enterprise
+
+    Returns:
+        Boolean
+
+    """
+    cache_key = get_cache_key(
```

Comment:
I have discovered the existence of https://github.com/edx/edx-platform/blob/master/openedx/core/lib/edx_api_utils.py#L17, which may be useful to us -- can you please review and decide if we can take advantage of the caching mechanism it provides?